testStrategies/commonToAllTestStrategies.go

```diff
@@ -360,13 +356,18 @@ func determineHostandPortforRequest(testDefinition *TestDefinition, configuratio
 	return targetHost, targetPort
 }
 
-func substituteRequestValues(requestBody *string, uniqueTestRunID string, globalsMap GlobalsMaps) string {
+func substituteRequestValues(requestBody *string, uniqueTestRunID string) string {
+	// Make a value copy for substitution.
 	requestPayloadCopy := *requestBody
 
-	//Get Global Properties for this test run
-	globalsMap.RLock()
-	testRunGlobals := globalsMap.m[uniqueTestRunID]
-	globalsMap.RUnlock()
+	// Lock global data structure for the duration of this function and
+	// those function that branch.
+	// TODO: Find a way to lock smaller sections. eg. how do we handle the "if GlobalsMap" statement?
+	mu.Lock()
+	defer mu.Unlock()
+
+	// Get the properties for this iteration.
+	testRunGlobals := GlobalsMap[uniqueTestRunID]
 
```

Comment:
Since we had to lock the whole function, I'd hoped the mutex would also apply to functions called from this function, but it doesn't work that way as we can see below. For example,